src/view.rs

```diff
@@ -155,13 +158,24 @@ pub fn view_menu_bar(m: &Model) -> Html {
         /* the "Nest Grid" button is special because
          * it contains fields for the variable size of the button
          */
-        <button class="menu-bar-button" id="nest" onclick=m.link.callback(move |_| {
-            if let Some(current) = &active_cell {
-                Action::AddNestedGrid(current.clone(), (default_row, default_col))
-            } else { Action::Noop }
-        })>
+        <button class="menu-bar-button" id="nest" 
+            onmousedown=m.link.callback(move |e : MouseDownEvent| {
+                if let Some(current) = &active_cell {
+                    Action::AddNestedGrid(current.clone(), (default_row, default_col))            
+                } else { Action::Noop }
+            })
+            onmouseup=m.link.callback(move |e : MouseUpEvent| {
+                if let Some(current) = &nest_active_cell.clone() {
+                    Action::SetActiveCell(current.clone())            
+                } else { Action::Noop }
```

Comment:
this is a good approach. could you add a comment explaining that this used to be an "OnClick", and that you changed it to this so two events could occur in sequence.